src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -32,11 +35,44 @@ private SslStreamCertificateContext(X509Certificate2 target, X509Certificate2[]
             IntermediateCertificates = intermediates;
             Trust = trust;
             SslContexts = new ConcurrentDictionary<SslProtocols, SafeSslContextHandle>();
+
+
+            using (RSAOpenSsl? rsa = (RSAOpenSsl?)target.GetRSAPrivateKey())
+            {
+                if (rsa != null)
+                {
+                    KeyHandle = rsa.DuplicateKeyHandle();
+                }
+            }
+
+            if (KeyHandle== null)
+            {
+                using (ECDsaOpenSsl? ecdsa = (ECDsaOpenSsl?)target.GetECDsaPrivateKey())
+                {
+                    if (ecdsa != null)
+                    {
+                        KeyHandle = ecdsa.DuplicateKeyHandle();
+                    }
+                }
+            }
+
+            if (KeyHandle== null)
+            {
+                throw new NotSupportedException(SR.net_ssl_io_no_server_cert);
+            }
+
+            CertificateHandle = Interop.Crypto.X509UpRef(target.Handle);
         }
 
         internal static SslStreamCertificateContext Create(X509Certificate2 target) =>
             Create(target, null, offline: false, trust: null, noOcspFetch: true);
 
+        ~SslStreamCertificateContext()
+        {
+            CertificateHandle?.Dispose();
+            KeyHandle?.Dispose();
+        }
```

Comment:
Why is this needed?  CertificateHandle and KeyHandle don't have their own finalizers?

Reply:
no specific reason. removed.
I was experimenting with the ideas since this is not Disposable. Leaving it to GC is probably best.